packages/Operators/test/tstNodeToNodeOperator.cpp

```diff
@@ -0,0 +1,196 @@
+//---------------------------------------------------------------------------//
+/*
+  Copyright (c) 2012, Stuart R. Slattery
+  All rights reserved.
+
+  Redistribution and use in source and binary forms, with or without
+  modification, are permitted provided that the following conditions are
+  met:
+
+  *: Redistributions of source code must retain the above copyright
+  notice, this list of conditions and the following disclaimer.
+
+  *: Redistributions in binary form must reproduce the above copyright
+  notice, this list of conditions and the following disclaimer in the
+  documentation and/or other materials provided with the distribution.
+
+  *: Neither the name of the University of Wisconsin - Madison nor the
+  names of its contributors may be used to endorse or promote products
+  derived from this software without specific prior written permission.
+
+  THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+  "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+  LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR
+  A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT
+  HOLDER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
+  SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
+  LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
+  DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
+  THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+  (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+  OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+*/
+//---------------------------------------------------------------------------//
+/*!
+ * \file   tstNodeToNodeOperator.cpp
+ * \author Stuart R. Slattery
+ * \brief  NodeToNode operator tests.
+ */
+//---------------------------------------------------------------------------//
+
+#include <iostream>
+#include <vector>
+#include <cmath>
+#include <sstream>
+#include <stdexcept>
+#include <limits>
+
+#include <DTK_MapOperatorFactory.hpp>
+#include <DTK_Point.hpp>
+#include <DTK_BasicGeometryManager.hpp>
+#include <DTK_Entity.hpp>
+#include <DTK_EntityCenteredField.hpp>
+#include <DTK_FieldMultiVector.hpp>
+
+#include "Teuchos_UnitTestHarness.hpp"
+#include "Teuchos_RCP.hpp"
+#include "Teuchos_Ptr.hpp"
+#include "Teuchos_Array.hpp"
+#include "Teuchos_ArrayRCP.hpp"
+#include "Teuchos_DefaultComm.hpp"
+#include "Teuchos_CommHelpers.hpp"
+#include "Teuchos_ParameterList.hpp"
+#include "Teuchos_XMLParameterListCoreHelpers.hpp"
+
+//---------------------------------------------------------------------------//
+// Test epsilon.
+//---------------------------------------------------------------------------//
+
+const double epsilon = 1.0e-8;
```

Comment:
I agree this check should be tightened in the node to node map - the results should be exact. I will change this.